sample-app/src/main/res/layout/fragment_flow_create_payment.xml

```diff
@@ -20,64 +20,123 @@
+        app:layout_constraintTop_toBottomOf="@id/snapPaymentRefResponse" />
 
     <com.google.android.material.textfield.TextInputLayout
-        android:id="@+id/nonSnapAmountLayout"
+        android:id="@+id/ebtCashAmountLayout"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:layout_margin="16dp"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/snapResponse">
+        app:layout_constraintTop_toBottomOf="@id/snapPaymentResponse">
 
         <com.google.android.material.textfield.TextInputEditText
-            android:id="@+id/nonSnapAmountEditText"
+            android:id="@+id/ebtCashAmountEditText"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:hint="Non Snap amount"
+            android:hint="EBT Cash amount or Payment ref"
             android:text="1" />
     </com.google.android.material.textfield.TextInputLayout>
 
-    <Button
-        android:id="@+id/submitNonSnapAmount"
+    <LinearLayout
+        android:id="@+id/ebtCashContainer"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_margin="16dp"
-        android:text="Submit Non Snap Amount"
+        android:layout_margin="12dp"
+        android:gravity="center"
+        android:orientation="horizontal"
+        android:padding="12dp"
+        app:layout_constraintBottom_toTopOf="@id/ebtCashResponse"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/ebtCashAmountLayout">
+
+        <Button
+            android:id="@+id/setEbtCashRef"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="8dp"
+            android:text="Set EBT Cash ref"
+            android:textSize="12sp" />
+
+        <Button
+            android:id="@+id/submitEbtCashAmount"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:text="Set EBT Cash Amount"
+            android:textSize="12sp" />
+    </LinearLayout>
+
+    <TextView
+        android:id="@+id/ebtCashPaymentRefResponse"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toTopOf="@id/ebtCashResponse"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/nonSnapAmountLayout" />
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/ebtCashContainer" />
 
     <TextView
-        android:id="@+id/nonSnapResponse"
+        android:id="@+id/ebtCashResponse"
```

Comment:
will we need to update mobile-qa-tests/ to support this?

Reply:
Yup, we will.